go/cmd/exporter/exporter.go

```diff
@@ -0,0 +1,170 @@
+// Package main runs the GCS exporter.
+// The exporter is responsible for creating many files in the OSV vulnerabilities bucket:
+// - [ECOSYSTEM]/VULN-ID.json - OSV JSON file for each vulnerability in each ecosystem
+// - [ECOSYSTEM]/all.zip - contains each OSV JSON file for that ecosystem
+// - [ECOSYSTEM]/modified_id.csv - contains the (modified, ID) of each vulnerability in the ecosystem directory
+// - /ecosystems.txt - a line-separated list of each exported ecosystem
+// - /all.zip - contains every OSV JSON file across all ecosytems
+// - /modified_id.csv - the (modified, [ECOSYSTEM]/ID) of every vulnerability across all ecosystem directories
+// - GIT/osv_git.json - a json array of every OSV vulnerability that has Vanir signatures.
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"cloud.google.com/go/storage"
+	"github.com/google/osv.dev/go/logger"
+	"github.com/ossf/osv-schema/bindings/go/osvschema"
+	"google.golang.org/api/iterator"
+)
+
+func main() {
+	logger.InitGlobalLogger()
+
+	outBucketName := flag.String("bucket", "osv-vulnerabilities", "Output bucket or directory name. If -local is true, this is a local path; otherwise, it's a GCS bucket name.")
+	vulnBucketName := flag.String("osv_vulns_bucket", os.Getenv("OSV_VULNERABILITIES_BUCKET"), "GCS bucket to read vulnerability protobufs from. Can also be set with the OSV_VULNERABILITIES_BUCKET environment variable.")
+	local := flag.Bool("local", true, "If true, writes the output to a local directory specified by -bucket instead of a GCS bucket.")
+	numWorkers := flag.Int("num_workers", 200, "The total number of concurrent workers to use for downloading from GCS and writing the output.")
+
+	flag.Parse()
+
+	logger.Info("exporter starting",
+		slog.String("bucket", *outBucketName),
+		slog.String("osv_vulns_bucket", *vulnBucketName),
+		slog.Bool("local", *local),
+		slog.Int("num_workers", *numWorkers))
+
+	if *vulnBucketName == "" {
+		logger.Fatal("OSV_VULNERABILITIES_BUCKET must be set")
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	cl, err := storage.NewClient(ctx)
+	if err != nil {
+		logger.Fatal("failed to create storage client", slog.Any("err", err))
+	}
+
+	vulnBucket := cl.Bucket(*vulnBucketName)
+	var outBucket *storage.BucketHandle
+	var outPrefix string
+	if *local {
+		outPrefix = *outBucketName
+	} else {
+		outBucket = cl.Bucket(*outBucketName)
+	}
+
+	objCh := make(chan *storage.ObjectHandle)
+	resultsCh := make(chan *osvschema.Vulnerability)
```

Comment:
Could be clearer to rename these channels like `<source>To<dist>Ch`. e.g. downloaderToRouterCh